src/Data/V/Linear/Internal/Ambiguous.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# OPTIONS_HADDOCK hide #-}
+
+module Data.V.Linear.Internal.Ambiguous (theLength, make, FunN) where
+
+import Data.Kind
+import Data.Proxy
+import Data.Type.Equality
+import {-# SOURCE #-} Data.V.Linear.Internal.V (V (..), cons)
+import qualified Data.Vector as Vector
+import GHC.Exts (proxy#)
+import GHC.TypeLits
+import Prelude.Linear.Internal
+import qualified Unsafe.Linear as Unsafe
+import Prelude
+  ( Bool (..),
+    Either (..),
+    Int,
+    Maybe (..),
+    error,
+    fromIntegral,
+    (-),
+  )
+
+theLength :: forall n. KnownNat n => Int
+theLength = fromIntegral (natVal' @n (proxy# @_))
+
+-- TODO: improve
+-------------------------------------------------------------------------------
```

Comment:
Improve what?

Reply:
Improve the `make` implementation, and remove all the Nat-related functions